client/lib/media/library-selected-store.js

```diff
@@ -84,9 +84,15 @@ MediaLibrarySelectedStore.getAll = function( siteId ) {
 		return [];
 	}
 
-	return MediaLibrarySelectedStore._media[ siteId ].map( function( itemId ) {
-		return MediaStore.get( siteId, itemId );
-	} );
+	// Avoid keeping invalid items in the selected list.
+	return (
+		MediaLibrarySelectedStore
+			._media[ siteId ]
+			.map( itemId => MediaStore.get( siteId, itemId ) )
+			.filter( ( item ) => {
+				return ( item && item.guid );
```

Comment:
Looks like it was a small regression. Basically if you uploaded from a url, it wouldn't be selected until the upload finished.